lib/addons/secrets-store/csi-driver-provider-aws.ts

```diff
@@ -0,0 +1,311 @@
+ */
+export interface KubernetesSecretData {
+
+  /**
+   * Name of the AWS Secret that is syncd
+   */
+  objectName: string;
+
+  /**
+   * Kubernetes Secret Key
+   */
+  key: string;
+}
+
+export enum AwsSecretType {
+  SSMPARAMETER = 'ssmparameter',
+  SECRETSMANAGER = 'secretsmanager'
+}
+
+export enum KubernetesSecretType {
+  OPAQUE = 'Opaque',
+  BASIC_AUTH = 'kubernetes.io/basic-auth',
+  TOKEN = 'bootstrap.kubernetes.io/token',
+  DOCKER_CONFIG_JSON = 'kubernetes.io/dockerconfigjson',
+  DOCKER_CONFIG = 'kubernetes.io/dockercfg',
+  SSH_AUTH = 'kubernetes.io/ssh-auth',
+  SERVICE_ACCOUNT_TOKEN = 'kubernetes.io/service-account-token',
+  TLS = 'kubernetes.io/tls'
+}
+
+const CsiDriverProviderAwsDefaults: CsiDriverProviderAwsProps = {
+  namespace: 'kube-system',
+  version: 'v0.0.23',
+  rotationPollInterval: undefined,
+  syncSecrets: true
+}
+
+export class CsiDriverProviderAws implements SecretsProvider {
+
+  private options: CsiDriverProviderAwsProps;
+  public name: string;
+
+  constructor(props?: CsiDriverProviderAwsProps) {
+    this.options = { ...CsiDriverProviderAwsDefaults, ...props };
+    this.name = Constants.SECRETS_STORE_CSI_DRIVER;
+  }
+
+  provide(clusterInfo: ClusterInfo): KubernetesManifest {
+    const cluster = clusterInfo.cluster;
+
+    type chartValues = {
+      linux: {
+        image: {
+          tag: string
+        }
+      },
+      enableSecretRotation?: string,
+      rotationPollInterval?: string,
+      syncSecret?: {
+        enabled: string
+      }
+    }
+
+    let values: chartValues = {
+      linux: {
+        image: {
+          tag: this.options.version!
+        }
+      }
+    };
+
+    if (typeof(this.options.rotationPollInterval) === 'string') {
+      values.enableSecretRotation = 'true';
+      values.rotationPollInterval = this.options.rotationPollInterval;
+    }
+
+    if (this.options.syncSecrets === true) {
+      values.syncSecret = {
+        enabled: 'true'
+      }
+    }
+
+    const secretStoreCSIDriverHelmChart = cluster.addHelmChart('SecretsStoreCSIDriver', {
+      chart: Constants.SECRETS_STORE_CSI_DRIVER,
+      repository: 'https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/master/charts',
+      namespace: this.options.namespace,
+      version: this.options.version,
+      release: Constants.SECRETS_STORE_CSI_DRIVER,
+      wait: true,
+      timeout: cdk.Duration.minutes(15),
+      values,
+    });
+
+    const manifestUrl = `https://raw.githubusercontent.com/aws/secrets-store-csi-driver-provider-aws/main/deployment/aws-provider-installer.yaml`;
+    const manifest: Record<string, any>[] = loadExternalYaml(manifestUrl);
+    const secretProviderManifest = clusterInfo.cluster.addManifest('SecretsStoreCsiDriverProviderAws', ...manifest);
+    secretProviderManifest.node.addDependency(secretStoreCSIDriverHelmChart);
+    return secretProviderManifest;
+  }
+}
+
+export class CsiDriverProviderAwsSecrets implements SecretsInfo {
```

Comment:
should we put this in a different file?

Reply:
Yeah I was thinking the same thing as I was reviewing it myself. Will do that.